src/reducers/accordionMenu.spec.tsx

```diff
@@ -0,0 +1,72 @@
+import reducer, { actions, initialState, isExpanded } from './accordionMenu';
+
+describe(__filename, () => {
+  describe('toggleItem', () => {
+    it('expands an item by default', () => {
+      const itemId = 'any item ID';
+
+      const state = reducer(undefined, actions.toggleItem({ itemId }));
+
+      expect(state.expanded).toEqual(itemId);
+    });
+
+    it('collapses an item if expanded', () => {
+      const itemId = 'any item ID';
+
+      let state;
+      // Expand it:
+      state = reducer(state, actions.toggleItem({ itemId }));
+      // This should collapse it:
+      state = reducer(state, actions.toggleItem({ itemId }));
+
+      expect(state.expanded).toEqual(null);
+    });
+
+    it('expands an item if collapsed', () => {
```

Comment:
Do we really need this case? The test above, for `collapses an item if expanded` proves that an item that was expanded will be collapsed when toggling, and that a collapsed item means that `state.expanded === null`. So this test is setting `state.expanded = null` (by expanding and collapsing an item), and then toggling it again, which is basically the same as the test for `expands an item by default`, which starts with `state.expanded === null` and then toggles an item and asserts that `state.expanded === itemId`.

I'm actually okay with the test, as I tend to like having more tests, but I feel like if I had written this test, and you were reviewing it, you might be saying the above to me about why it's not needed. 😉

Reply:
It might be a little overkill but I added this explicit test to differentiate the starting value from that of `initialState` since technically they are different code paths.

Reply:
To be clear, it is executing a unique code path. It's making sure that the *reducer* resets the state before expanding again.